modules/http4s/src/smithy4s/http4s/EndpointSpecificMiddleware.scala

```diff
@@ -0,0 +1,39 @@
+package smithy4s
+package http4s
+
+import org.http4s.HttpApp
+
+// format: off
+trait EndpointSpecificMiddleware[Alg[_[_, _, _, _, _]], F[_]] {
+  def prepare(service: Service[Alg])(
+      endpoint: Endpoint[service.Operation, _, _, _, _, _]
+  ): HttpApp[F] => HttpApp[F]
+}
+// format: on
+
+object EndpointSpecificMiddleware {
+
+  trait Simple[Alg[_[_, _, _, _, _]], F[_]]
+      extends EndpointSpecificMiddleware[Alg, F] {
```

Comment:
@Baccata I know I could use `Kind5.Existential` here instead of passing the Alg, but then I am unable to pass that simple middleware to the restJson builder middleware function since it expects an Alg for the service it is building for. Any ideas of how I can get around that?

Reply:
Doesn't the code I shared above compile as is ? Any is kind-polymorphic

Reply:
Ergh, doesn't look like it is in Scala 3

Reply:
Yes, move the `Alg` type parameter from the trait to the method.